packages/atlas/src/components/_inputs/WhiteListTextField/WhiteListTextField.tsx

```diff
@@ -0,0 +1,118 @@
+import { useApolloClient } from '@apollo/client'
+import debouncePromise from 'awesome-debounce-promise'
+import React, { useRef, useState } from 'react'
+
+import { GetMembershipsDocument, GetMembershipsQuery, GetMembershipsQueryVariables } from '@/api/queries'
+import { Avatar } from '@/components/Avatar'
+import { SvgActionCancel } from '@/components/_icons'
+import { createLookup } from '@/utils/data'
+import { SentryLogger } from '@/utils/logs'
+
+import { MemberBadgesWrapper, StyledMemberBadge, StyledSelectedText } from './WhiteListTextField.styles'
+
+import { TextFieldWithDropdown } from '../TextFieldWithDropdown'
+
+export type Member = {
+  id: string
+  handle?: string | null
+  avatarUri?: string | null
+}
+
+type WhiteListTextFieldProps = {
+  selectedMembers: Member[]
+  setSelectedMembers: React.Dispatch<React.SetStateAction<Member[]>>
+}
+
+export const WhiteListTextField: React.FC<WhiteListTextFieldProps> = ({ selectedMembers, setSelectedMembers }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [members, setMembers] = useState<Member[]>([])
+  const client = useApolloClient()
+  const [isError, setIsError] = useState(false)
+
+  const debounceFetchMembers = useRef(
+    debouncePromise(async (val?: string) => {
+      if (!val) {
+        setMembers([])
+        return
+      }
+      try {
+        const {
+          data: { memberships },
+        } = await client.query<GetMembershipsQuery, GetMembershipsQueryVariables>({
+          query: GetMembershipsDocument,
+          variables: { where: { handle_startsWith: val } },
+        })
+        setIsLoading(false)
+
+        const members = memberships.map(({ handle, id, metadata: { avatar } }) => ({
+          handle,
+          avatarUri: avatar?.__typename === 'AvatarUri' ? avatar.avatarUri : undefined,
```

Comment:
This will not work if the avatar is uploaded through the storage system right? I think we should handle that case as well to be future-proof